crates/uv-client/src/mac_version.rs

```diff
@@ -0,0 +1,36 @@
+use platform_tags::PlatformError;
+use serde::Deserialize;
+
+/// Get the macOS version from the SystemVersion.plist file.
+pub(crate) fn get_mac_os_version() -> Result<String, PlatformError> {
```

Comment:
Do we need this? I thought we got it from the interpreter now.

Reply:
Oh that's the same thing, thanks https://github.com/astral-sh/uv/pull/2509

Reply:
Note, it's not quite just major.minor, it should include the product version from the plist as-is based on my testing with pip.

Reply:
Happy to hand this off to a mac user who can test this on an actual mac